src/components/AddressFormFields.tsx

```diff
@@ -1,172 +1,149 @@
+  const [states, setStates] = useState([]);
+
+  useEffect(() => {
+    if(Object.keys(address).length !== 0 && address.country !== "")
+    {
+      let alpha2 = getCountryFromName(address.country).alpha2.toLowerCase();
+
+      let newStates = getStates(alpha2).map(state => { 
+        return {value: state, label: state}
+      })
+      setStates(newStates);
+    }
+  }, [address.country])
+
+  useEffect(() => {
+    let baseCountries = getCountries();
+
+    let newCountries = baseCountries.map(country => 
+    {
+
+      return {value: country.code, label: country.name}
+    })
+
+
+    setCountries(newCountries);
+  }, [])
+
+  const handleGoogleAddressChange = (value, action) => {
+    if(action.action === "clear")
+    {
+      setGoogleAddress({})
+    }
+    else 
+    setGoogleAddress(value);
+  }
+
+  const addressChange = (key) => {
+    if(key !== null && key !== "")
+      address.setAddress(key);
+  }
+
   return (
     <div style={{ display: "flex", flexDirection: "column" }}>
-      <Form.Row
-        style={{
-          display: "flex",
-          flexDirection: "row",
-        }}
-      >
-        <Form.Group
-          as={Col}
-          controlId="formGridName"
-          style={{ marginRight: "2rem" }}
-        >
-          <Form.Label>Name</Form.Label>
-          <Form.Control
-            type="Name"
-            placeholder="Name"
-            required={true}
-            onChange={(event: any) => {
-              props.setName(event.target.value);
-            }}
-            disabled={props.loading}
-          />
-        </Form.Group>
 
-        <Form.Group as={Col} controlId="formGridEmail">
-          <Form.Label>Email</Form.Label>
-          <Form.Control
-            type="email"
-            placeholder="Enter email"
-            required={true}
-            onChange={(event: any) => {
-              props.setEmail(event.target.value);
-            }}
-            disabled={props.loading}
-          />
-        </Form.Group>
-      </Form.Row>
+      <div style={{ display: "flex", flexDirection: "row" }}>
+
+          <div className="halfInputBox1">
+            <Form.Control
+              type="Name"
+              style={{height: "37px"}}
+              placeholder="First Name"
+              required={true}
+              onChange={e => address.setFirstName(e.target.value)}
+              disabled={props.loading}
+            />
+          </div>
 
-      <Form.Group controlId="formGridAddress1">
-        <Form.Label>Address</Form.Label>
+          <div className="halfInputBox2">
+            <Form.Control
+              style={{height: "37px"}}
+              type="Name"
+              placeholder="Last Name"
+              required={true}
+              onChange={e => address.setLastName(e.target.value)}
+              disabled={props.loading}
+            />
+          </div>
+      </div>
+
+      <GooglePlacesAutocomplete
+        apiKey="AIzaSyB587mOW5-3gNFOSPJeGuJhqf0FUj910uM"
```

Comment:
need to move this to the env variables. i think you have access to the vercel settings but lmk if you want me to add it up there

Reply:
done